crates/matrix-sdk-crypto/src/session_manager/sessions.rs

```diff
@@ -225,16 +225,21 @@ impl SessionManager {
         let mut missing: BTreeMap<_, BTreeMap<_, _>> = BTreeMap::new();
         let mut timed_out: BTreeMap<_, BTreeSet<_>> = BTreeMap::new();
 
-        // Add the list of devices that the user wishes to establish sessions
-        // right now.
-        for user_id in users.filter(|u| !self.failures.contains(u.server_name())) {
-            let user_devices = Box::pin(
-                self.key_request_machine
-                    .identity_manager()
-                    .get_user_devices_for_encryption(user_id),
-            )
-            .await?;
+        let unfailed_users = users.filter(|u| !self.failures.contains(u.server_name()));
 
+        // XXX: You'd think this would be unnecessary, but if you omit it, Rust starts
+        // yelling about lifetimes and Send for reasons that I don't really understand.
+        let unfailed_users: Vec<_> = unfailed_users.collect();
```

Comment:
You can remove this if you change `impl IntoIterator` in `impl Iterator` in the signature of `get_user_devices_for_encryption`.

Reply:
Of course I can. Thanks for that, I was totally blanking on it.